mitiq/pec/representations/learning.py

```diff
@@ -15,16 +15,109 @@
 """Function to calculate parameters for biased noise model via a
 learning-based technique."""
 
-from typing import Optional, Dict, Any, List
+from typing import cast, Optional, Dict, Any, List, Tuple
 import numpy as np
+from scipy.optimize import minimize
 from mitiq import QPROGRAM, Executor, Observable
+from mitiq.cdr import generate_training_circuits
 from mitiq.pec import execute_with_pec
 from mitiq.pec.representations.biased_noise import (
     represent_operation_with_local_biased_noise,
 )
 
 
-def _biased_noise_loss_function(
+def learn_depolarizing_noise_parameter(
+    operations_to_learn: List[QPROGRAM],
+    circuit: QPROGRAM,
+    ideal_executor: Executor,
+    noisy_executor: Executor,
+    pec_kwargs: Dict["str", Any],
+    num_training_circuits: int = 5,
+    fraction_non_clifford: float = 0.2,
+    training_random_state: np.random.RandomState = None,
+    epsilon0: float = 0.05,
+    observable: Optional[Observable] = None,
+    **minimize_kwargs: Dict["str", Any],
+) -> Tuple[bool, float]:
+    r"""This function learns the depolarizing noise parameter (epsilon)
+    associated to a set of input operations. The learning process is based on
+    the execution of a set of training circuits on a noisy backend and on a
+    classical simulator. The training circuits are near-Clifford approximations
+    of the input circuit. A depolarizing noise model characterization is
+    assumed.
```

Comment:
What about if we mention somewhere in the docstring something like this?

_Please note that using this function is quite difficult. One of the main difficulties is setting a good value of `num_samples` in the PEC options (`pec_kwargs`).  Setting a small value of `num_samples` is typically necessary to obtain a reasonable execution time. On the other hand, with a small number of PEC samples, the statistical error can be so large that the optimization process fails._

There is probably a better way to rephrase this. But it can be worth giving a similar warning to user before they go through the same problems that we had in our CI tests.

Reply:
Good point, I added the following paragraph just after the text you marked?

> Note: using this function may require some tuning. One of the main
> challenges is setting a good value of `num_samples` in the PEC options
> `pec_kwargs`. Setting a small value of `num_samples` is typically
>  necessary to obtain a reasonable execution time. On the other hand,
>  using a number of PEC samples that is too small can result in a large
>  statistical error, ultimately causing the optimization process to fail.